sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -117,49 +115,67 @@ object TypeCoercion {
    * loss of precision when widening decimal and double, and promotion to string.
    */
   private[analysis] def findWiderTypeForTwo(t1: DataType, t2: DataType): Option[DataType] = {
-    (t1, t2) match {
-      case (t1: DecimalType, t2: DecimalType) =>
-        Some(DecimalPrecision.widerDecimalType(t1, t2))
-      case (t: IntegralType, d: DecimalType) =>
-        Some(DecimalPrecision.widerDecimalType(DecimalType.forType(t), d))
-      case (d: DecimalType, t: IntegralType) =>
-        Some(DecimalPrecision.widerDecimalType(DecimalType.forType(t), d))
-      case (_: FractionalType, _: DecimalType) | (_: DecimalType, _: FractionalType) =>
-        Some(DoubleType)
-      case _ =>
-        findTightestCommonTypeToString(t1, t2)
-    }
+    findTightestCommonType(t1, t2)
+      .orElse(findWiderTypeForDecimal(t1, t2))
```

Comment:
yea we changed the order, but looks like it won't change the result. `findWiderTypeForDecimal` will always return a result for decimal type and numeric type, and if `findTightestCommonType` can return a result, `findWiderTypeForDecimal` will return the same result. So it doesn't matter if we run `findTightestCommonType` before or after it.